lib/collection/src/shards/local_shard_operations.rs

```diff
@@ -73,6 +73,12 @@ impl LocalShard {
                     .unwrap()
                     .distance;
                 let processed_res = vector_res.into_iter().map(|mut scored_point| {
+                    // Don't post-process if we are dealing with custom scoring
+                    // TODO(luis): add discovery variant here too
+                    if matches!(req.query, QueryEnum::RecommendBestScore(_)) {
+                        return scored_point;
+                    }
+
```

Comment:
What about doing an exhaustive match here, so that we don't require a TODO?

Reply:
Much better